### PR TITLE
fix: tests reliability

### PR DESCRIPTION
**Pull request by ganchoradkov**
# Description
PR to improve tests reliability.
Changes:

- remove trailing `/` on relay URLs as per relay specs
- update socket disconnect logic to use `transportClose`
- relay tests to run in a single thread synchronously
- removes unnecessary pairing flows in some tests to reduce overhead

## How Has This Been Tested?
npm run test
CI runs

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: packages/sign-client/package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
     "test:concurrency": "vitest run --dir test/concurrency",
     "test:xregion": "vitest run --dir test/xregion",
     "test": "npm run test:pre; npm run test:run",
-    "test:ignoreUnhandled": "npm run test:pre; npm run test:run -- --dangerouslyIgnoreUnhandledErrors",
+    "test:ignoreUnhandled": "npm run test:pre; npm run test:run -- --dangerouslyIgnoreUnhandledErrors --no-threads",
```
Comment (arein) [inline packages/sign-client/package.json:30]: how much slower is this making tests?

**Review by arein — APPROVED** [review]

File: packages/sign-client/package.json (hunk 1 above)
Comment (ganchoradkov) [reply packages/sign-client/package.json:30]: Comparing previous CI runs

- [3m 21s](https://github.com/WalletConnect/rs-relay/actions/runs/3450693108/jobs/5778391203) - old
- [4m 15s](https://github.com/WalletConnect/rs-relay/actions/runs/3574722287/jobs/6012511070) - new
I would say ~22% but on some endpoints the diff is marginal

File: packages/core/test/shared/ws.ts [hunk 2]
```diff
@@ -1,18 +1,7 @@
-import { IJsonRpcConnection } from "@walletconnect/jsonrpc-utils";
 import { IRelayer } from "@walletconnect/types";
-import EventEmitter from "events";
 
 export async function disconnectSocket(relayer: IRelayer) {
   if (relayer.connected) {
-    relayer.provider.events = new EventEmitter();
```
Comment (bkrem) [inline packages/core/test/shared/ws.ts:7]: Glad we can get rid of this hack 🙌

File: packages/sign-client/test/sdk/validation.spec.ts [hunk 3]
```diff
@@ -29,7 +29,6 @@ describe("Sign Client Validation", () => {
     let clients: Clients;
     beforeAll(async () => {
       clients = await initTwoClients();
-      await testConnectMethod(clients);
```
Comment (bkrem) [inline packages/sign-client/test/sdk/validation.spec.ts:32]: Nice optimisation ✨ Seems obvious in hindsight that the validation tests don't need any existing connection but never noticed this.

**Review by bkrem — APPROVED** [review]

**Pull request merged**